ext/strings_test.go

```diff
@@ -329,3 +330,266 @@ func TestStringsWithExtension(t *testing.T) {
+			format:         "%s%%",
+			formatArgs:     `"percent on the right"`,
+			expectedOutput: "percent on the right%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the left",
+			format:         "%%%s",
+			formatArgs:     `"percent on the left"`,
+			expectedOutput: "%percent on the left",
+		},
+		{
+			name:           "multiple substitutions",
+			format:         "%d %d %d, %s %s %s, %d %d %d, %s %s %s",
+			formatArgs:     `1, 2, 3, "A", "B", "C", 4, 5, 6, "D", "E", "F"`,
+			expectedOutput: "1 2 3, A B C, 4 5 6, D E F",
+		},
+		{
+			name:           "percent sign escape sequence support",
+			format:         "\u0025\u0025escaped \u0025s\u0025\u0025",
+			formatArgs:     `"percent"`,
+			expectedOutput: "%escaped percent%",
+		},
+		{
+			name:           "fixed point formatting clause",
+			format:         "%.3f",
+			formatArgs:     "1.2345",
+			expectedOutput: "1.234",
+		},
+		{
+			name:           "binary formatting clause",
+			format:         "this is 5 in binary: %b",
+			formatArgs:     "5",
+			expectedOutput: "this is 5 in binary: 101",
+		},
+		{
+			name:           "octal formatting clause",
+			format:         "%o",
+			formatArgs:     "11",
+			expectedOutput: "13",
+		},
+		{
+			name:           "lowercase hexadecimal formatting clause",
+			format:         "%h is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1e is 20 in hexadecimal",
+		},
+		{
+			name:           "uppercase hexadecimal formatting clause",
+			format:         "%H is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1E is 20 in hexadecimal",
+		},
+		{
+			name:           "string support with hexadecimal formatting clause",
+			format:         "%h",
+			formatArgs:     `"Hello world!"`,
+			expectedOutput: "48656c6c6f20776f726c6421",
+		},
+		{
+			name:           "string support with uppercase hexadecimal formatting clause",
+			format:         "%H",
+			formatArgs:     `"Hello world!"`,
+			expectedOutput: "48656C6C6F20776F726C6421",
+		},
+		{
+			name:           "byte support with hexadecimal formatting clause",
+			format:         "%h",
+			formatArgs:     `b"byte string"`,
+			expectedOutput: "6279746520737472696e67",
+		},
+		{
+			name:           "byte support with uppercase hexadecimal formatting clause",
+			format:         "%H",
+			formatArgs:     `b"byte string"`,
+			expectedOutput: "6279746520737472696E67",
+		},
+		{
+			name:           "lowercase scientific notation formatting clause",
+			format:         "%.6e",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f\u00d7\u202f10\u2070\u00b3",
+		},
+		{
+			name:           "uppercase scientific notation formatting clause",
+			format:         "%.6E",
+			formatArgs:     "1052.032911275",
+			expectedOutput: "1.052033\u202f\u00d7\u202f10\u2070\u00b3",
+		},
+		{
+			name:           "locale support",
+			format:         "%.3f",
+			formatArgs:     "3.14",
+			locale:         "fr_FR",
```

Comment:
Shouldn't most (all?) tests have a variant that uses the non-default locale?

Reply:
Yes,  though it's worth pointing out that only the output of `%e` and `%f` are locale-dependent. It'd still be good to ensure other clauses remain unchanged regardless of locale.

Reply:
> It'd still be good to ensure other clauses remain unchanged regardless of locale.

The is the primary motivation for my statement. It should be easy to test.

Reply:
(marking as resolved now that i18n-independent tests have an extra check to ensure they don't change with locale)